source/d3d12/d3d12_device.cpp

```diff
@@ -158,15 +379,47 @@ HRESULT STDMETHODCALLTYPE D3D12Device::CreateComputePipelineState(const D3D12_CO
 }
 HRESULT STDMETHODCALLTYPE D3D12Device::CreateCommandList(UINT nodeMask, D3D12_COMMAND_LIST_TYPE type, ID3D12CommandAllocator *pCommandAllocator, ID3D12PipelineState *pInitialState, REFIID riid, void **ppCommandList)
 {
-	return _orig->CreateCommandList(nodeMask, type, pCommandAllocator, pInitialState, riid, ppCommandList);
+	const HRESULT hr = _orig->CreateCommandList(nodeMask, type, pCommandAllocator, pInitialState, riid, ppCommandList);
+
+	if (FAILED(hr))
+	{
+		LOG(WARN) << "> 'ID3D12Device::CreateCommandList' failed with error code " << std::hex << hr << std::dec << "!";
+		return hr;
+	}
+
+	if (riid == __uuidof(ID3D12GraphicsCommandList) || riid == __uuidof(ID3D12GraphicsCommandList1))
+	{
+		if (_runtimes.empty())
+			return hr;
+
+		// retrieve current runtime
+		const auto runtime = _runtimes.front();
+		d3d12_current_runtime = runtime.get();
+		d3d12_current_device = this;
```

Comment:
This is rather dangerous, since the app can technically create a commandlist on a different device before recording into the one just created here, which will set `d3d12_current_device` to something wrong. Would be safer to have a global lookup table (`std::unordered_map<ID3D12CommandList *, D3D12Device *>`) to map from commandlist to the matching device and runtime.

Reply:
I understand. From what i have tested now, it seems not to raise pbs, but in the future, it probably will...

Reply:
In Shadow Of the Tomb Raider, it is more than necessary